From 24bc82e0f7c5943bd2513ce648e875dc290eab7f Mon Sep 17 00:00:00 2001
From: Watson Ladd <watsonbladd@gmail.com>
Date: Thu, 27 Apr 2023 10:16:49 -0700
Subject: [PATCH 153/381] Clarify documentation of SSL_SESSION_dup

Reviewed-by: Matt Caswell <matt@openssl.org>
Reviewed-by: Todd Short <todd.short@me.com>
Reviewed-by: Tomas Mraz <tomas@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/20848)

(cherry picked from commit 25bab273ccc9b517cc4c1783950e3f95421cb570)
---
 doc/man3/SSL_SESSION_free.pod | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/doc/man3/SSL_SESSION_free.pod b/doc/man3/SSL_SESSION_free.pod
index 74c6ec2a29..33475d76b6 100644
--- a/doc/man3/SSL_SESSION_free.pod
+++ b/doc/man3/SSL_SESSION_free.pod
@@ -21,8 +21,8 @@ SSL_SESSION_free - create, free and manage SSL_SESSION structures
 SSL_SESSION_new() creates a new SSL_SESSION structure and returns a pointer to
 it.
 
-SSL_SESSION_dup() copies the contents of the SSL_SESSION structure in B<src>
-and returns a pointer to it.
+SSL_SESSION_dup() creates a new SSL_SESSION structure that is a copy of B<src>.
+The copy is not owned by any cache that B<src> may have been in.
 
 SSL_SESSION_up_ref() increments the reference count on the given SSL_SESSION
 structure.
@@ -62,6 +62,8 @@ incorrect reference counts and therefore program failures.
 SSL_SESSION_new returns a pointer to the newly allocated SSL_SESSION structure
 or NULL on error.
 
+SSL_SESSION_dup returns a pointer to the new copy or NULL on error.
+
 SSL_SESSION_up_ref returns 1 on success or 0 on error.
 
 =head1 SEE ALSO
-- 
2.34.1

